### PR TITLE
Qml components traces feature

### PR DESCRIPTION
**Pull request by pasnox**

### CODE REVIEW

**Comment by jpetersen** [comment]
Hi,

looks great in general, I noticed in QQ2 gallery example that something is wrong with the paint order when I have the menu overlay on the left open.

**Comment by pasnox** [comment]
Indeed, I can reproduce, will fix !
Looks like the issue comes with popup or window overlay.

**Comment by jpetersen** [comment]
Looks good

File: plugins/quickinspector/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ if(Qt5Quick_FOUND AND HAVE_PRIVATE_Qt5Quick_HEADERS AND NOT ${Qt5Quick_VERSION}
   include_directories(SYSTEM ${Qt5Quick_PRIVATE_INCLUDE_DIRS})
 
   add_library(gammaray_quickinspector_shared STATIC ${gammaray_quickinspector_shared_srcs})
-  target_link_libraries(gammaray_quickinspector_shared gammaray_common Qt5::Gui Qt5::Quick)
+  target_link_libraries(gammaray_quickinspector_shared gammaray_common gammaray_core Qt5::Gui Qt5::Quick)
```
Comment (vkrause) [inline plugins/quickinspector/CMakeLists.txt]: This links probe code into the client, that is dangerous as that makes assumptions of running in target everywhere. Can we avoid this?

File: plugins/quickinspector/quickitemgeometry.cpp [hunk 2]
```diff
@@ -112,6 +142,10 @@ void QuickItemGeometry::initFrom(QQuickItem *item)
         QQuickItemPrivate *parentPriv = QQuickItemPrivate::get(parent);
         parentTransform = parentPriv->itemToWindowTransform();
     }
+
+    traceColor = colorForItem(item);
+    traceTypeName = ObjectDataProvider::typeName(item);
+    traceName = ObjectDataProvider::name(item);
```
Comment (vkrause) [inline plugins/quickinspector/quickitemgeometry.cpp]: This seems to be the only dependency on probe code in the shared bit, can probably be addressed by moving this to the caller instead.

File: plugins/quickinspector/CMakeLists.txt (hunk 1 above)
Comment (pasnox) [inline plugins/quickinspector/CMakeLists.txt]: Good catch, will try

File: plugins/quickinspector/quickitemgeometry.cpp (hunk 2 above)
Comment (pasnox) [inline plugins/quickinspector/quickitemgeometry.cpp]: Yeah the ObjectDataProvider was the only required thing from the gammaray_core.
What do you mean by moving this to the caller ?

**Comment by pasnox** [comment]
@vkrause Split the overlay/decoration drawing stuff now.

**Pull request merged**